File: src/wzh/varcheck/parse/Location.java
package wzh.varcheck.parse;

public class Location implements Cloneable {
    int line;
    int column;

    Location(int line, int column) {
        this.line = line;
        this.column = column;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        return super.clone();
    }

    void newLine() {
        column = 0;
        line++;
    }

    void shift() { column++; }
}

File: src/wzh/varcheck/parse/Location.kt
package wzh.varcheck.parse

class Location (val line: Int, val column: Int) : Cloneable {

    constructor() : this(0, 0 )

    internal fun newLine(): Location {
        return Location(line + 1, 0)
    }

    internal fun shift(): Location {
        return Location(line, column + 1)
    }

    override fun toString(): String {
        return "Location{$line, $column}"
    }

}
